src/lineup/provider/TDPLocalDataProvider.ts

```diff
@@ -0,0 +1,31 @@
+import {IColumnConstructor, IColumnDesc, IDataProviderOptions, ILocalDataProviderOptions, isSupportType, ITypeFactory, LocalDataProvider} from 'lineupjs';
+
+/**
+ * A data provider which changes the default column width from LineUp
+ */
+export default class TDPLocalDataProvider extends LocalDataProvider {
+  constructor(_data: any[], columns: IColumnDesc[] = [], options: Partial<ILocalDataProviderOptions & IDataProviderOptions> = {}) {
+    super(_data, columns, options);
+  }
+
+  protected instantiateColumn(type: IColumnConstructor, id: string, desc: IColumnDesc, typeFactory: ITypeFactory) {
+    // cache the column width because initializing the `type` class mutates the desc object
+    const columnWidth = desc.width;
+
+    // create a column instance needed for the `isSupportType(col)`
+    const col = new type(id, desc, typeFactory);
+
+    // do nothing if column width is already defined or it is a support type column (e.g., rank, aggregation, selection)
+    if (columnWidth >= 0 || isSupportType(col)) {
```

Comment:
What if the column itself defines a default (as in https://github.com/lineupjs/lineupjs/blob/develop/src/model/SelectionColumn.ts#L73-L75), but is no support type? Then it would override the default value of the column because we are only looking at the incoming desc, not the "outcoming" width value?
I am not sure if that ever occurs in LineUp, but I think we should check all columns first if this combination occurs anywhere before merging this PR. If it appears it will be much more complicated to achieve properly...

Reply:
Thanks for pointing this edge case out! 👍 My last commit should fix it. Please check again.

![grafik](https://user-images.githubusercontent.com/5851088/116992705-be436a00-acd6-11eb-94ec-1db9f5094ef7.png)

Reply:
Looks good to me.